data/transactions/logic/eval.go

```diff
@@ -1866,9 +1875,10 @@ func opBytesEq(cx *EvalContext) error {
 		return errors.New("math attempted on large byte-array")
 	}
 
-	rhs := new(big.Int).SetBytes(cx.stack[last].Bytes)
-	lhs := new(big.Int).SetBytes(cx.stack[prev].Bytes)
-	cx.stack[prev] = boolToSV(lhs.Cmp(rhs) == 0)
+	rhs := nonzero(cx.stack[last].Bytes)
+	lhs := nonzero(cx.stack[prev].Bytes)
```

Comment:
I guess this can be replaced by `bytes.TrimLeft(..., string([]byte{0})`?

Reply:
Yes, but I have strong reason to believe that would be slower and allocate.


// TrimLeft returns a subslice of s by slicing off all leading
// UTF-8-encoded code points contained in cutset.
func TrimLeft(s []byte, cutset string) []byte {
	return TrimLeftFunc(s, makeCutsetFunc(cutset))
}

(and `makeCutSetFunction` returns a closure, so it needs to allocate)

Reply:
I see, I wasn't going into details of the bytes library implementation.

But still one question, will it make a difference in perf by returning `[]` or returning `nil` at the end of `nonzero`?

Reply:
I had to return something that was logically equivalent to the empty slice at the end of the byte slice given.  That might have argued for something like `b[len(b):len(b)]`.  That might be legal (not sure if the first index can be `len(b)`, but seems more confusing.  And I didn't want to make a new byte slice with `[]byte{}` as I believe that must allocate to given you a backing array. (But maybe not, if it sets cap==0, I suppose it might not give you a backing array at all.)

Anyway, the usual Go recommendation for an empty slice is `nil`.

Reply:
got it, that makes sense to me